cmd/habitat/community/state/transitions.go

```diff
@@ -211,3 +223,158 @@ func (t *AddNodeTransition) Validate(oldState *community.CommunityState) error {
 
 	return nil
 }
+
+type StartProcessTransition struct {
+	Process *community.Process
+}
+
+func (t *StartProcessTransition) Type() string {
+	return TransitionTypeStartProcess
+}
+
+func (t *StartProcessTransition) Patch(oldState *community.CommunityState) ([]byte, error) {
+	marshaledProcess, err := json.Marshal(t.Process)
+	if err != nil {
+		return nil, err
+	}
+
+	return []byte(fmt.Sprintf(`[{
+		"op": "add",
+		"path": "/processes/-",
+		"value": %s
+	}]`, string(marshaledProcess))), nil
+}
+
+func (t *StartProcessTransition) Validate(oldState *community.CommunityState) error {
+	// make sure no process with same process id and app name is already started
+	for _, p := range oldState.Processes {
+		if p.ID == t.Process.ID {
+			return fmt.Errorf("process ID %s already used in community %s", t.Process.ID, oldState.CommunityID)
+		}
+		if p.AppName == t.Process.AppName {
+			return fmt.Errorf("community %s already has app %s running", oldState.CommunityID, t.Process.ID)
+		}
+	}
+
+	return nil
+}
+
+type StopProcessTransition struct {
+	ProcessID string
+}
+
+func (t *StopProcessTransition) Type() string {
+	return TransitionTypeStopProcess
+}
+
+func (t *StopProcessTransition) Patch(oldState *community.CommunityState) ([]byte, error) {
+	for i, p := range oldState.Processes {
+		if p.ID == t.ProcessID {
+			return []byte(fmt.Sprintf(`[{
+				"op": "remove",
+				"path": "/processes/%d"
+			}]`, i)), nil
+		}
+	}
+
+	return nil, fmt.Errorf("process ID %s not found in processes list", t.ProcessID)
+}
+
+func (t *StopProcessTransition) Validate(oldState *community.CommunityState) error {
+	// ensure no process_instances with the process ID are still around
+	for _, i := range oldState.ProcessInstances {
+		if i.ProcessID == t.ProcessID {
+			return fmt.Errorf("there are still process instances with process ID %s", t.ProcessID)
+		}
+	}
+
+	for _, p := range oldState.Processes {
+		if p.ID == t.ProcessID {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("process ID %s not found in processes list", t.ProcessID)
+}
+
+type StartProcessInstanceTransition struct {
+	ProcessInstance *community.ProcessInstance
```

Comment:
i think i thought `StartProcess` would be what's here and there's no such thing as `StartProcessInstance`. but we need something that makes all the non-process-running nodes aware of the processes right? maybe we should rename `StartProcess` to something more explicit about this like `AddRunningProcess` or `AddCommunityApp`

Reply:
I think this might make more sense based off my comment about `ProcessInstances` mapping nodes to `Processes`. Lmk if you are still confused about this. Can also write up some docs for this stuff.

Reply:
yeah i think docs would be good. comment above makes sense, but now i'm wondering what's the difference between a process and an app? i guess processes don't necessarily need to be apps but the opposite isn't true.